src/decisionengine/framework/logicengine/FactLookup.py

```diff
@@ -20,23 +20,28 @@ class FactLookup:
 
     As an example, consider the following configuration:
 
-      facts: {
-        should_publish: "(True)",
-      },
-      rules: {
-        publish_1: {
-          expression: "should_publish",
-          facts: ["should_publish"]
-        },
-        publish_2: {
-          expression: "should_publish",
-          actions: ["go_to_press"]
-          facts: ["should_publish"]
+    .. code-block:: json
+
+        {
+            "facts": {
+                "should_publish": "(True)"
+            },
+            "rules": {
+                "publish_1": {
+                    "expression": "should_publish",
+                    "facts": ["should_publish"]
+                },
+                "publish_2": {
+                    "expression": "should_publish",
+                    "actions": ["go_to_press"],
+                    "facts": ["should_publish"]
+                },
+                "retract": {
+                    "expression": "not should_publish",
+                    "facts": ["should_retract"]
+                }
+            }
         }
```

Comment:
I rewrote some Jsonnet code blocks on docstrings as JSON because Sphinx doesn't seem to support Jsonnet. An alternative solution would be using text code blocks instead.